include/fmt/ranges.h

```diff
@@ -247,31 +247,48 @@ template <typename Range>
 using value_type =
     remove_cvref_t<decltype(*detail::range_begin(std::declval<Range>()))>;
 
-template <typename Arg, FMT_ENABLE_IF(!is_like_std_string<
-                                      typename std::decay<Arg>::type>::value)>
-FMT_CONSTEXPR const char* format_str_quoted(bool add_space, const Arg&) {
-  return add_space ? " {}" : "{}";
+template <typename OutputIt, typename Formatting>
+OutputIt add_range_formatting_spaces(OutputIt out,
+                                     const Formatting& formatting) {
+  if (const_check(formatting.add_prepostfix_space)) {
+    *out++ = ' ';
+  }
+  out = detail::copy(formatting.delimiter, out);
+  if (const_check(formatting.add_delimiter_spaces)) {
+    *out++ = ' ';
+  }
+  return out;
 }
 
-template <typename Arg, FMT_ENABLE_IF(is_like_std_string<
-                                      typename std::decay<Arg>::type>::value)>
-FMT_CONSTEXPR const char* format_str_quoted(bool add_space, const Arg&) {
-  return add_space ? " \"{}\"" : "\"{}\"";
+template <
+    typename Char, typename OutputIt, typename Arg,
+    FMT_ENABLE_IF(std::is_same<Arg, const char*>::value ||
```

Comment:
What is `std::is_same<Arg, const char*>::value` for?

Reply:
I thought that this was required to make it so that for example a `const char * array_of_literals[]` would be formatted with `"`s, but I appear to be incorrect. I've added a test-case to cover that and you're correct, it appears unnecessary, though I couldn't say why.